File: 06-june/22-maximum-number-of-balloons/solution.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        counts = [0] * 26
        for char in text:
            counts[ord(char) - ord("a")] += 1

        return min(
            counts[ord("b") - ord("a")],
            counts[ord("a") - ord("a")],
            counts[ord("l") - ord("a")] // 2,
            counts[ord("o") - ord("a")] // 2,
            counts[ord("n") - ord("a")],
        )

File: 06-june/22-maximum-number-of-balloons/solution.java
class Solution {
    public int maxNumberOfBalloons(String text) {
        int[] counts = new int[26];
        for (int i = 0; i < text.length(); i++) {
            counts[text.charAt(i) - 'a']++;
        }

        return Math.min(
            Math.min(counts['b' - 'a'], counts['a' - 'a']),
            Math.min(
                Math.min(counts['l' - 'a'] / 2, counts['o' - 'a'] / 2),
                counts['n' - 'a']
            )
        );
    }
}
